arangod/Pregel/PregelFeature.cpp

```diff
@@ -316,12 +316,16 @@ void PregelFeature::cleanupWorker(uint64_t executionNumber) {
 
 void PregelFeature::cleanupAll() {
   MUTEX_LOCKER(guard, _mutex);
-  _conductors.clear();
-  for (auto it : _workers) {
+  decltype(_conductors) cs = std::move(_conductors);
+  decltype(_workers) ws = std::move(_workers);
+  guard.unlock();
+    
+  cs.clear();
```

Comment:
mmh. `cs` is a copy/move of `_conductors`, but is instantly cleared afterwards. Is that intentional?

Reply:
Looks like it. The old code simply cleared it under the lock. This way the deallocation can be done outside the lock, while the move itself should likely be quicker.